packages/mattermost-redux/src/selectors/entities/preferences.ts

```diff
@@ -0,0 +1,277 @@
+
+import {createShallowSelector} from 'mattermost-redux/utils/helpers';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+import {setThemeDefaults} from 'mattermost-redux/utils/theme_utils';
+
+export function getMyPreferences(state: GlobalState): { [x: string]: PreferenceType } {
+    return state.entities.preferences.myPreferences;
+}
+
+export function get(state: GlobalState, category: string, name: string, defaultValue: any = '') {
+    const key = getPreferenceKey(category, name);
+    const prefs = getMyPreferences(state);
+
+    if (!(key in prefs)) {
+        return defaultValue;
+    }
+
+    return prefs[key].value;
+}
+
+export function getBool(state: GlobalState, category: string, name: string, defaultValue = false): boolean {
+    const value = get(state, category, name, String(defaultValue));
+    return value !== 'false';
+}
+
+export function getInt(state: GlobalState, category: string, name: string, defaultValue = 0): number {
+    const value = get(state, category, name, defaultValue);
+    return parseInt(value, 10);
+}
+
+export function makeGetCategory(): (state: GlobalState, category: string) => PreferenceType[] {
+    return createSelector(
+        getMyPreferences,
+        (state: GlobalState, category: string) => category,
+        (preferences, category) => {
+            const prefix = category + '--';
+            const prefsInCategory: PreferenceType[] = [];
+
+            for (const key in preferences) {
+                if (key.startsWith(prefix)) {
+                    prefsInCategory.push(preferences[key]);
+                }
+            }
+
+            return prefsInCategory;
+        },
+    );
+}
+
+const getDirectShowCategory = makeGetCategory();
+
+export function getDirectShowPreferences(state: GlobalState) {
+    return getDirectShowCategory(state, Preferences.CATEGORY_DIRECT_CHANNEL_SHOW);
+}
+
+const getGroupShowCategory = makeGetCategory();
+
+export function getGroupShowPreferences(state: GlobalState) {
+    return getGroupShowCategory(state, Preferences.CATEGORY_GROUP_CHANNEL_SHOW);
+}
+
+const getFavoritesCategory = makeGetCategory();
+
+export function getFavoritesPreferences(state: GlobalState) {
+    const favorites = getFavoritesCategory(state, Preferences.CATEGORY_FAVORITE_CHANNEL);
+    return favorites.filter((f) => f.value === 'true').map((f) => f.name);
+}
+
+export const getVisibleTeammate: (state: GlobalState) => Array<$ID<UserProfile>> = createSelector(
+    getDirectShowPreferences,
+    (direct) => {
+        return direct.filter((dm) => dm.value === 'true' && dm.name).map((dm) => dm.name);
+    },
+);
+
+export const getVisibleGroupIds: (state: GlobalState) => string[] = createSelector(
+    getGroupShowPreferences,
+    (groups) => {
+        return groups.filter((dm) => dm.value === 'true' && dm.name).map((dm) => dm.name);
+    },
+);
+
+export const getTeammateNameDisplaySetting: (state: GlobalState) => string = createSelector(
```

Comment:
The return value of this used to be nullable which the web app used to be fine with before. Now that we've merged the repos, it started complaining about type issues, so I made it not nullable and added a default value below.